tests/10-issue.js

```diff
@@ -2,119 +2,260 @@
+            if (credentialStatusType === 'string') {
+              issuedVc.credentialStatus.type = [issuedVc.credentialStatus.type];
+            }
+            issuedVc.credentialStatus.type.should.include.members([
+              'BitstringStatusListEntry'
+            ]);
           });
-      });
-    }
-  });
-
-describe('BitstringStatusList Credentials (Issue "statusPurpose: suspension")',
-  function() {
-    this.matrix = true;
-    this.report = true;
-    this.implemented = [...match.keys()];
-    this.rowLabel = 'Test Name';
-    this.columnLabel = 'Issuer';
-    for(const [issuerName, {issuers}] of match) {
-      describe(issuerName, function() {
-        let issuerResponse;
-        let err;
-        let issuedVc;
-        before(async function() {
-          const issuer = issuers.find(issuer => issuer.tags.has('Suspension'));
-          const {result, error, data} = await issueVc({issuer});
-          err = error;
-          issuerResponse = result;
-          issuedVc = data;
-        });
-        it('MUST issue a VC with BitstringStatusList "credentialStatus" and ' +
-          '"suspension" status purpose', async function() {
-          this.test.cell = {columnId: issuerName, rowId: this.test.title};
-          should.exist(issuerResponse);
-          should.not.exist(err);
-          issuerResponse.status.should.equal(201);
-          should.exist(issuedVc, `Expected VC from ${issuerName} to exist.`);
-          testCredential({credential: issuedVc});
+          it('The purpose of the status entry MUST be a string.',
+            async function () {
+              this.test.cell = { columnId: issuerName, rowId: this.test.title };
+              issuedVc.credentialStatus.should.contain.keys('statusPurpose');
+              issuedVc.credentialStatus.statusPurpose.should.be.a('string');
+            });
+          it('While the value of the string is arbitrary, ' +
+            'the following values MUST be used for their intended purpose',
+            async function () {
+              if (issuedVc.credentialStatus.statusPurpose == 'revocation') {
+                this.test.cell = { columnId: issuerName, rowId: this.test.title };
+              } else if (issuedVc.credentialStatus.statusPurpose === 'suspension') {
+                this.test.cell = { columnId: issuerName, rowId: this.test.title };
+              } else if (issuedVc.credentialStatus.statusPurpose === 'message') {
+                this.test.cell = { columnId: issuerName, rowId: this.test.title };
+              } else {
+                this.skip()
+              };
+            });
+          it('The statusListIndex property MUST be an arbitrary size integer ' +
+            'greater than or equal to 0, expressed as a string in base 10.',
+            async function () {
+              this.test.cell = { columnId: issuerName, rowId: this.test.title };
+              issuedVc.credentialStatus.should.contain.keys('statusListIndex');
```

Comment:
this might be overparanoid, but don't forget about Optional Chanining: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining

I use it to much, but it could help in some situations.